core/state_processor.go

```diff
@@ -154,7 +154,10 @@ func ProcessParentBlockHash(prevHash common.Hash, vmenv *vm.EVM) {
 	msg := types.NewMessage(consensus.SystemAddress, &params.HistoryStorageAddress, 0, common.Big0, 30_000_000, common.Big0, common.Big0, common.Big0, prevHash.Bytes(), nil, false, nil, nil)
 	vmenv.Reset(NewEVMTxContext(msg), vmenv.StateDB)
 	vmenv.StateDB.AddAddressToAccessList(params.HistoryStorageAddress)
-	_, _, _ = vmenv.Call(vm.AccountRef(msg.From()), *msg.To(), msg.Data(), 30_000_000, common.Big0)
+	_, _, err := vmenv.Call(vm.AccountRef(msg.From()), *msg.To(), msg.Data(), 30_000_000, common.Big0)
+	if err != nil {
+		log.Error("Failed to store parent block hash in history storage contract", "err", err)
```

Comment:
go-ethereum panics here https://github.com/ethereum/go-ethereum/blob/cd78b65cdaa9e00866962e38685c4f93ac9e8444/core/state_processor.go#L258

but IDK why

Reply:
Panic is risky I think. The data cannot be inconsistent as there will be check for merkle trie root in core/block_validator.go:ValidateState